core/src/test/java/org/apache/iceberg/view/ParserTestBase.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.view;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.apache.iceberg.util.JsonUtil;
+import org.apache.iceberg.util.TestJsonUtil;
+import org.junit.Assert;
+import org.junit.Test;
+
+public abstract class ParserTestBase<T> {
+
+  private final T entry;
+  private final String json;
+  private final JsonUtil.JsonWriter<T> writer;
+  private final JsonUtil.JsonReader<T> reader;
+
+  public ParserTestBase(
```

Comment:
I am wondering if we really should structure the tests like this with a test base. Each object model could have different fields to be tested and this makes the tests not as flexible. Also we added all the things in the JsonUtil which I am not sure if it is useful outside this context

Reply:
Agreed, I think what we can do is just follow the existing pattern for testing table metadata and just enumerate different toJson/fromJson cases. While the benefit of having a parent class is to reduce some duplication, for the parsing code it seems better just to have more explicit tests (for example like we do with ref parser)